File: problems/219/solution.go
package main
func containsNearbyDuplicate(nums []int, k int) bool {
    for i := 0; i < len(nums); i++ {
        for j := max(0, i - k); j < i; j++ {
            if nums[i] == nums[j] {
                return true
            }
        }
    }
    
    return false;
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}

File: problems/219/solution.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> um;

        for (int i = 0; i < size(nums); i++) {
            if (um.count(nums[i])) {
                if (i - um[nums[i]] <= k) {
                    return true;
                }
            }

            um[nums[i]] = i;
        }

        return false;
    }
};